vars/testsPublishResults.groovy

```diff
@@ -0,0 +1,220 @@
+        // archive results
+        archiveResults(settings.get('archive'), pattern, allowEmpty)
+    }
+}
+
+def publishCoberturaReport(Map settings = [:]) {
+    if(settings.active){
+        def pattern = settings.get('pattern')
+        def allowEmpty = settings.get('allowEmptyResults')
+
+        cobertura(
+        //step([
+        //    $class: 'CoberturaPublisher',
+            coberturaReportFile: pattern,
+            onlyStable: settings.get('onlyStableBuilds'),
+            failNoReports: !allowEmpty,
+            failUnstable: false,
+            failUnhealthy: false,
+            autoUpdateHealth: false,
+            autoUpdateStability: false,
+            maxNumberOfBuilds: 0
+        )
+        // archive results
+        archiveResults(settings.get('archive'), pattern, allowEmpty)
+    }
+}
+
+// publish Performance Report using "Jenkins Performance Plugin" https://wiki.jenkins.io/display/JENKINS/Performance+Plugin
+def publishJMeterReport(Map settings = [:]){
+    if(settings.active){
+        def pattern = settings.get('pattern')
+
+        step([
+            $class: 'PerformancePublisher',
+            errorFailedThreshold: settings.get('errorFailedThreshold'),
+            errorUnstableThreshold: settings.get('errorUnstableThreshold'),
+            errorUnstableResponseTimeThreshold: settings.get('errorUnstableResponseTimeThreshold'),
+            relativeFailedThresholdPositive: settings.get('relativeFailedThresholdPositive'),
+            relativeFailedThresholdNegative: settings.get('relativeFailedThresholdNegative'),
+            relativeUnstableThresholdPositive: settings.get('relativeUnstableThresholdPositive'),
+            relativeUnstableThresholdNegative: settings.get('relativeUnstableThresholdNegative'),
+            modePerformancePerTestCase: false,
+            modeOfThreshold: settings.get('modeOfThreshold'),
+            modeThroughput: settings.get('modeThroughput'),
+            nthBuildNumber: settings.get('nthBuildNumber'),
+            configType: settings.get('configType'),
+            failBuildIfNoResultFile: settings.get('failBuildIfNoResultFile'),
+            compareBuildPrevious: settings.get('compareBuildPrevious'),
+            parsers: asList(getJMeterParser().newInstance(pattern))
+        ])
+        // archive results
+        archiveResults(settings.get('archive'), pattern, settings.get('allowEmptyResults'))
+    }
+}
+
+def touchFiles(){
+    echo "[${STEP_NAME}] update test results"
+    def patternArray = pattern.split(',')
+    for(def i = 0; i < patternArray.length; i++){
+        sh "find . -wholename '${patternArray[i].trim()}' -exec touch {} \\;"
+    }
+}
+
+@NonCPS
+def getJMeterParser(){
+    // handle package renaming of JMeterParser class
+    try {
+        return this.class.classLoader.loadClass("hudson.plugins.performance.parsers.JMeterParser")
+    } catch (Exception e) {
+        return this.class.classLoader.loadClass("hudson.plugins.performance.JMeterParser")
```

Comment:
@marcusholl @OliverNocon any idea how to mock away the class loader stuff here?

Reply:
I think we should be able to get rid of it completely.
This was due to a incompatible change in the plugin way back.

This problem should not exist anymore and we could call the step directly.

Reply:
But we still need to load one or the other class I think.